File: mobile/rescufy/lib/domain/entities/register_request.dart
// lib/domain/entities/register_request.dart
import 'package:equatable/equatable.dart';

class RegisterRequest extends Equatable {
  final String fullName;
  final String email;
  final String password;
  final String nationalId;
  final String phoneNumber;
  final int age;
  final String gender;

  const RegisterRequest({
    required this.fullName,
    required this.email,
    required this.password,
    required this.nationalId,
    required this.phoneNumber,
    required this.age,
    required this.gender,
  });

  @override
  List<Object?> get props => [
    fullName,
    email,
    password,
    nationalId,
    phoneNumber,
    age,
    gender,
  ];
}

File: RescufyBackendNew/Shared/DTOs/Auth/RegisterRequest.cs
using Microsoft.AspNetCore.Http;

namespace Shared.DTOs.Auth
{
    public class RegisterRequest
    {
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string UserName { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;

        [System.ComponentModel.DataAnnotations.Required]
        [System.ComponentModel.DataAnnotations.StringLength(14, MinimumLength = 14, ErrorMessage = "National ID must be exactly 14 digits")]
        [System.ComponentModel.DataAnnotations.RegularExpression("^[0-9]*$", ErrorMessage = "National ID must be numeric")]
        public string NationalId { get; set; } = default!;

        [System.ComponentModel.DataAnnotations.Required]
        [System.ComponentModel.DataAnnotations.RegularExpression("^(Male|Female)$", ErrorMessage = "Gender must be 'Male' or 'Female'")]
        public string Gender { get; set; } = default!;

        [System.ComponentModel.DataAnnotations.Required]
        [System.ComponentModel.DataAnnotations.Range(1, 120, ErrorMessage = "Age must be between 1 and 120")]
        public int Age { get; set; }
        
        public IFormFile? ProfileImage { get; set; }
    }
}
